lib/HttpClient.ts

```diff
@@ -259,96 +273,136 @@ export class HttpClient {
         let usingSsl = info.parsedUrl.protocol === 'https:';
         info.httpModule = usingSsl ? https : http;
         var defaultPort: number = usingSsl ? 443 : 80;
+        info.options = <http.RequestOptions>{};
```

Comment:
It would be great if this PR proactively had contextual clues for the reader. It looks like at least some of the diff is related to code moving around, which is hard to interpret.

Reply:
I agree unfortunately github PR experience isn't great to review such changes, I have tried to add comments where I though they would help. Mostly the change is in the _getAgent function which is used to get agent, rest of the info properties remain the same. In _getAgent we are either returning one of the private agents (_agent or _proxyAgent) depending on reuse or creating a new agent. If you need more info about any particular change I can help.